src/components/userComponents/emailPopup/page.tsx

```diff
@@ -0,0 +1,198 @@
+'use client';
+
+import React, { Fragment, useState } from 'react';
+import { Dialog, Transition } from '@headlessui/react';
+import { PiPaperPlaneTiltBold , PiSealCheck } from "react-icons/pi";
+import { VscClose } from "react-icons/vsc";
+import { BiErrorCircle } from "react-icons/bi";
+
+import Link from 'next/link';
+
+
+/**
+ * @returns an email pop up
+ */
+export default function EmailPopup(
+ ) {
+  const [isOpen, setIsOpen] = useState(true);
+  const [inputValue, setInputValue] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [subscribed, setSubscribed] = useState(false);
+
+
+  const closeModal = () => {
+    setIsOpen(false);
+  }
+
+  const handleChange = (e) => {
+    setInputValue(e.target.value);
+    setShowError(false);
+  };
+
+  const isValidEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.(com|ca)$/;
+    return emailRegex.test(email);
+  };
+
```

Comment:
feat: so that we hit all potential email edge cases, could you add support for the following email suffixes.
.edu: Education institutions
.gov: Government entities
.org: Non-profit organizations
.net: Network-related organizations